### DISCUSSION TITLE
One big store or bunch of small stores

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [pmndrs/valtio#745] [Q&A] [answered]
I am making React Native app and I want to dodge the context shenanigans and use this library instead.

I am wondering if I should create bunch of small stores or one big one. Would there be any differences? If I understand "proxying" correctly, then there should be no difference if I have two stores with array of 10k items, one for news, one for notifications, and useSnapshot them both or just one store with total of 20k items.

**Comment by dai-shi** [comment, marked as answer]
Should mostly be the same.

Does this help?
https://github.com/pmndrs/valtio/blob/5d0eca7edc25aa7f17fb8082ef14244a35fc12b2/docs/how-tos/how-to-split-and-compose-states.mdx#L7-L39
